packages/composite-checkout/src/components/billing-fields.js

```diff
@@ -0,0 +1,557 @@
+						type="checkbox"
+						id="domain-registration"
+						name="domain-registration"
+						defaultChecked={ ! showDomainContactFields }
+						onChange={ toggleDomainFieldsVisibility }
+					/>
+					<DomainRegistrationLabel htmlFor="domain-registration">
+						{ localize(
+							'Use your billing details for your domain registration contact information.'
+						) }
+					</DomainRegistrationLabel>
+				</DomainRegistrationCheckBoxWrapper>
+			) }
+
+			{ showDomainContactFields && hasDomainsInCart( items ) && (
+				<DomainFields paymentData={ paymentData } setPaymentData={ setPaymentData } />
+			) }
+		</BillingFormFields>
+	);
+}
+
+const BillingFormFields = styled.div`
+	margin-bottom: 16px;
+`;
+
+const FormField = styled( Field )`
+	margin-top: 16px;
+	:first-child {
+		margin-top: 0;
+	}
+`;
+
+const FieldRow = styled( GridRow )`
+	margin-top: 16px;
+
+	first-child {
+		margin-top: 0;
+	}
+`;
+
+function hasDomainsInCart( items ) {
+	if ( items.find( item => item.type === 'domain' ) ) {
+		return true;
+	}
+
+	return false;
+}
+
+function isElligibleForVat() {
+	//Detect whether people are in EU or AU and return true if they are
+	return false;
+}
+
+function DomainFieldsCheckbox( toggleVisibility ) {
+	return (
+		<DomainRegistrationCheckBoxWrapper>
+			<DomainRegistrationCheckbox
+				type="checkbox"
+				id="domain-registration"
+				name="domain-registration"
+				defaultChecked={ true }
+				onChange={ toggleVisibility }
+			/>
+			<DomainRegistrationLabel htmlFor="domain-registration">
+				Use your billing details for your domain registration contact information.
+			</DomainRegistrationLabel>
+		</DomainRegistrationCheckBoxWrapper>
+	);
+}
+
+DomainFieldsCheckbox.propTypes = {
+	toggleVisibility: PropTypes.func,
+};
+
+const DomainRegistrationCheckBoxWrapper = styled.div`
+	margin: 16px 0 24px;
+	display: flex;
+	width: 100%;
+`;
+
+const DomainRegistrationLabel = styled.label`
+	font-size: 14px;
+	color: ${props => props.theme.colors.textColor};
+	display: block;
+	position: relative;
+	padding-left: 5px;
+
+	:hover {
+		cursor: pointer;
+	}
+
+	:before {
+		display: block;
+		width: 16px;
+		height: 16px;
+		content: '';
+		position: absolute;
+		left: -20px;
+		top: 0;
+		border: 1px solid ${props => props.theme.colors.borderColorLight};
+		border-radius: 3px;
+	}
+`;
+
+const DomainRegistrationCheckbox = styled.input`
+	margin-right: 5px;
+	opacity: 0;
+
+	:checked + label:before {
+		background: ${props => props.theme.colors.highlight};
+	}
+
+	:checked + label:after {
+		display: block;
+		width: 4px;
+		height: 8px;
+		content: '';
+		position: absolute;
+		left: -14px;
+		top: 3px;
+		border-right: 2px solid ${props => props.theme.colors.surface};
+		border-bottom: 2px solid ${props => props.theme.colors.surface};
+		transform: rotate( 45deg );
+	}
+
+	:focus + label:before {
+		outline: ${props => props.theme.colors.outline} auto 5px;
+	}
+`;
+
+function AddressFields( { paymentData, setPaymentData, location } ) {
```

Comment:
I think that the variable name `location` might be misleading here. Maybe `addressType`?